### PR TITLE
Refactor nuke to use timers

### PR DESCRIPTION
**Pull request by forestbbbbbbb**
## About The Pull Request

Refactors the nuke to use timers, it shouldn't take twice the time set to explode anymore (i think)

## Changelog

:cl:
refactor: nuke uses timers
/:cl:

### CODE REVIEW

**Comment by ClosetedSkeletonGH** [comment]
1 nuke second being 2 real seconds was bogus. Ive never once seen marines lose a nuke hold ever with the extra 3 minutes tacked onto it, this wont change anything imo

File: code/game/objects/machinery/nuclearbomb.dm [hunk 1]
```diff
@@ -49,41 +52,31 @@
...
-		return
-	timeleft--
-	if(timeleft <= 0)
-		explode()
-		return
-	updateUsrDialog()
-
-/obj/machinery/nuclearbomb/start_processing()
-	. = ..()
+/obj/machinery/nuclearbomb/proc/enable()
```
Comment (ivanmixo) [inline code/game/objects/machinery/nuclearbomb.dm:55]: Autodoc

File: code/game/objects/machinery/nuclearbomb.dm [hunk 2]
```diff
@@ -49,41 +52,31 @@
...
 	countdown.start()
 	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_NUKE_START, src)
+	notify_ghosts("[usr] enabled the [src], it has [time] seconds on the timer.", source = src, action = NOTIFY_ORBIT, extra_large = TRUE)
+	timer_enabled = TRUE
+	timer = addtimer(CALLBACK(src, PROC_REF(explode)), time SECONDS, TIMER_STOPPABLE)
 	update_minimap_icon()
-	notify_ghosts("[usr] enabled the [src], it has [timeleft] seconds on the timer.", source = src, action = NOTIFY_ORBIT, extra_large = TRUE)
 
-/obj/machinery/nuclearbomb/stop_processing()
+/obj/machinery/nuclearbomb/proc/disable()
```
Comment (ivanmixo) [inline code/game/objects/machinery/nuclearbomb.dm:64]: Autodoc

File: code/game/objects/machinery/nuclearbomb.dm [hunk 3]
```diff
@@ -20,11 +20,14 @@
...
-	var/timeleft = 180
+	///Time to start the timer on
+	var/time = 180
+	///Min time for the nuke timer
+	var/timemin = 180
 	///Max time for the nuke timer
-	var/timemax = 180
+	var/timemax = 600
 	var/timer_enabled = FALSE
+	var/timer
```
Comment (ivanmixo) [inline code/game/objects/machinery/nuclearbomb.dm:30]: Autodoc

File: code/game/objects/machinery/nuclearbomb.dm [hunk 4]
```diff
@@ -20,11 +20,14 @@
 	var/deployable = TRUE
 	var/extended = FALSE
 	var/lighthack = FALSE
-	///Time currently left on the nuke
-	var/timeleft = 180
+	///Time to start the timer on
+	var/time = 180
```
Comment (ivanmixo) [inline code/game/objects/machinery/nuclearbomb.dm:24]: Add a unit like `SECONDS` to this var define as well.

**Review by ivanmixo — CHANGES_REQUESTED** [review]
Not sure about the balance implications of nuke taking half as long as before. 

Double it so we retain old behavior and atomise this PR so it can be discussed separately.

**Review by ivanmixo — APPROVED** [review]
LGTM

**Pull request merged**